consent/api/v1/views.py

```diff
@@ -0,0 +1,214 @@
+# -*- coding: utf-8 -*-
+"""
+A generic API for edX Enterprise's Consent application.
+"""
+
+from __future__ import absolute_import, unicode_literals
+
+from consent.errors import ConsentAPIRequestError
+from consent.helpers import (
+    consent_exists,
+    consent_provided,
+    consent_required,
+    get_enterprise_course_enrollment,
+    get_enterprise_customer,
+    get_enterprise_customer_user,
+    get_user_id,
+)
+from rest_framework.response import Response
+from rest_framework.status import HTTP_200_OK, HTTP_400_BAD_REQUEST, HTTP_404_NOT_FOUND
+from rest_framework.views import APIView
+
+# Will be using internal models at a later time.
+from enterprise.models import EnterpriseCourseEnrollment
+
+
+class DataSharingConsentView(APIView):
+    """
+        **Use Cases**
+
+            Presents a generic data sharing consent API to applications
+            that have Enterprise customers who require data sharing
+            consent from users.
+
+        **Behavior**
+
+            Implements GET, POST, and DELETE which each have roughly
+            the following behavior (see their individual handlers for
+            more documentation):
+
+            GET /consent/api/v1/data_sharing_consent?username=bob&enterprise_customer=ENTERPRISE-UUID&course_id=ID
+            >>> {
+            >>>     "username": "bob",
+            >>>     "course_id": "course-v1:edX+DemoX+Demo_Course",
+            >>>     "enterprise_customer": "enterprise-uuid-goes-right-here",
+            >>>     "exists": False,
+            >>>     "consent_provided": False,
+            >>>     "consent_required": True,
+            >>> }
+
+            If the "exists" key is false, then the body will be returned
+            with a 404 Not Found error code; otherwise, 200 OK. If either
+            of enterprise_customer or username is not provided, an appropriate
+            400-series error will be returned.
+
+            POST or DELETE /consent/api/v1/data_sharing_consent
+            >>> {
+            >>>     "username": "bob",
+            >>>     "course_id": "course-v1:edX+DemoX+Demo_Course",
+            >>>     "enterprise_customer": "enterprise-uuid-goes-right-here"
+            >>> }
+
+            The API accepts JSON objects with these key-value pairs for
+            POST or DELETE.
```

Comment:
I don't think we should be doing that, see https://stackoverflow.com/questions/2539394/rest-http-delete-and-parameters for some reasons. I also tried it, and it broke a lot of tests, probably because this isn't really how it's meant to be used. I don't think we should have `DELETE` depend on query parameters.